packages/store/src/index.ts

```diff
@@ -0,0 +1,231 @@
+import type { JSX } from "solid-js";
+import { batch, untrack } from "solid-js";
+import type { SetStoreFunction } from "solid-js/store";
+import { createStore } from "solid-js/store";
+
+type AnyFunction = (...args: any[]) => any;
+type Action<A extends AnyFunction> = (...args: Parameters<A>) => ReturnType<A>;
+type CreateAction = <A extends AnyFunction>(fn: A) => Action<A>;
+type CreateActions = <A extends object | undefined>(fns?: A) => A | undefined;
+
+export const createAction: CreateAction =
+  fn =>
+  (...args) =>
+    batch(() => untrack(() => fn(...args)));
+export const createActions: CreateActions = actions => {
+  if (!actions) return undefined;
+
+  let result = { ...actions };
+  // for (const [name, fn] of Object.entries(result) as [keyof typeof result, AnyFunction][]) {
+  for (const [name, fn] of Object.entries(result)) {
+    if (typeof fn !== "function") continue;
+    result = {
+      ...result,
+      [name]: createAction(fn)
+    };
```

Comment:
No point in destructuring the `results` object here. It only makes the operation slower.
```suggestion
    result[name] = createAction(fn)
```